core/services/relay/evm/evm.go

```diff
@@ -518,12 +522,27 @@ func (r *Relayer) NewMedianProvider(rargs commontypes.RelayArgs, pargs commontyp
 	if err != nil {
 		return nil, err
 	}
-	return &medianProvider{
+
+	medianProvider := medianProvider{
 		configWatcher:       configWatcher,
 		reportCodec:         reportCodec,
 		contractTransmitter: contractTransmitter,
 		medianContract:      medianContract,
-	}, nil
+	}
+
+	chainReader, err := NewChainReaderService(lggr, r.chain.LogPoller(), contractID, &relayConfig)
+	if err != nil {
+		if errors.Is(err, errors.ErrUnsupported) {
+			// ignore for now, until we can remove old MedianContract code from MedianProvider
+			lggr.Info("ChainReader missing from RelayConfig; falling back to internal MedianContract")
```

Comment:
We can't assume that `errors.ErrUnsupported` always means `ChainReader missing from RelayConfig`. Why don't we just do that check before calling the constructor, and have the constructor accept only what it needs: `ChainReader`.